src/Switcher.js

```diff
@@ -81,47 +81,26 @@ function Switcher(props) {
   useEffect(() => {
     const handler = () => handleRouteChange.current();
     const usingProvider = Boolean(switcherProvider);
-    if (!usingProvider) {
-      load && window.addEventListener('load', handler);
-      pushState && window.addEventListener('popstate', handler);
-      hashChange && window.addEventListener('hashchange', handler);
+    if (usingProvider) {
```

Comment:
this inversion makes the code so much better! I love it.

What do you think of ALWAYS requiring the provider? such that the second case is never even possible?

I think lots of other libraries do this. Might be worth making some of these breaking changes before the next release.

Reply:
it is kind of nice to be able to throw in one component but i think i agree it doesn't scale well if i have multiple switchers rendered in one tree. but i'm not sure how often that case would occur.

Reply:
cool, we won't do that in this PR, but maybe before the release